src/functions.date.js

```diff
@@ -108,24 +108,52 @@ for (let unit in s) {
 	}, {multiValued: true});
 }
 
-_.duration = $.extend(function(ms) {
-	var count = ms || 0;
-	var unit = "ms";
-
-	for (let nextUnit in s) {
-		var nextCount = _.msTo(nextUnit, ms);
+_.duration = $.extend(function (ms, terms) {
+	if (ms === 0 || terms === undefined) {
+		terms = 1;
+	}
 
-		if (nextCount === 0) {
-			break;
+	let timeLeft = ms || 0;
+	let ret = [];
+	let units = Object.keys(s).reverse();
+	units.push("ms");
+	let unitsUsed = [];
+	let onlyConsecutiveNonzeros = true;
+
+	units.map(function(unit) {
+		// get largest value of time unit for the remaining
+		// time to account for
+		let unitValue = unit === "ms"? timeLeft : _.msTo(unit, timeLeft);
+		timeLeft -= unitValue * (unit === "ms" ? 1 : Mavo.Functions[unit]());
+		return unitValue;
+	}).forEach(function (currentValue, index) {
```

Comment:
If you're not using `map()`'s return value, what's the point of the subsequent `forEach()`? Why not use a single looping construct (whether that's `map()`, `forEach()`, `for .. of` loop or something else)?